packages/salesforcedx-vscode-replay-debugger/src/breakpoints/checkpointService.ts

```diff
@@ -53,13 +100,152 @@ export abstract class BaseNode extends TreeItem {
 
 export class CheckpointNode extends BaseNode {
   private readonly children: CheckpointInfoNode[] = [];
+  // These variables are necessary for the ApexExecutionOverlayAction
+  // and are not user editable. All user editable checkpoint pieces
+  // are stored on the children.
+  public executableEntityName: string;
+  public isDumpingHeap: boolean;
+  public line: number;
+  // Stored for convenience
+  public isTrigger: boolean;
+  // Source and URI are two things that the user cannot edit and
+  // it's debatable whether or not they need to be stored at all
+  public source: DebugProtocol.Source;
+  public uri: string;
+
+  constructor(
+    sourceInput: DebugProtocol.Source,
+    lineInput: number,
+    uriInput: string
+  ) {
+    super(
+      sourceInput.name + ':' + lineInput,
+      TreeItemCollapsibleState.Expanded
+    );
+
+    this.line = lineInput;
+    this.source = sourceInput;
+    this.uri = uriInput;
+    const sourceNameSplit = this.source.name!.split('.');
+    // Get the executableEntityName from the sourceInput, if this is a trigger
+    // then ensure that the SFDC_TRIGGER prefix is prepended to the trigger name
+    if (sourceNameSplit[1].toLocaleLowerCase() === 'trigger') {
+      this.executableEntityName = SFDC_TRIGGER + sourceNameSplit[0];
+      this.isTrigger = true;
+    } else {
+      this.executableEntityName = sourceNameSplit[0];
+      this.isTrigger = false;
+    }
+    this.isDumpingHeap = true;
+
+    // Create the items that the user is going to be able to control (Type, Script, Iteration)
+    const cpASTNode = new CheckpointInfoActionScriptTypeNode(
+      ActionScriptType.None
+    );
+    this.children.push(cpASTNode);
+    const cpScriptNode = new CheckpointInfoActionScriptNode('');
+    this.children.push(cpScriptNode);
+    const cpIterationNode = new CheckpointInfoIterationNode(1);
+    this.children.push(cpIterationNode);
+  }
+
+  public createJsonStringForApexExecutionOverlayAction(): string {
```

Comment:
I'm thinking you can have a type for ApexExecutionOverlayAction with fields like ExecutableEntityName, ActionScript, etc. Then you can create an instance of that object with the values that you have. With that, you can do `JSON.stringify(theCheckpointObject)` to get the JSON string instead of this string builder.

Reply:
I can probably do that, but I'm still going to have to go through an piece together the object. The CheckpointNode contains the pieces that aren't editable by the user. The list of children are the CheckpointInfoNode variants which contain the editable pieces. I'm still gong to need the createJsonStringForApexExecutionOverlayAction or some function to call that'll do this.